x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/aggregation/RateIntAggregator.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.aggregation;
+
+import org.apache.lucene.util.Accountable;
+import org.apache.lucene.util.RamUsageEstimator;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.util.ObjectArray;
+import org.elasticsearch.compute.ann.GroupingAggregator;
+import org.elasticsearch.compute.ann.IntermediateState;
+import org.elasticsearch.compute.data.Block;
+import org.elasticsearch.compute.data.BlockFactory;
+import org.elasticsearch.compute.data.DoubleBlock;
+import org.elasticsearch.compute.data.DoubleVector;
+import org.elasticsearch.compute.data.IntBlock;
+import org.elasticsearch.compute.data.IntVector;
+import org.elasticsearch.compute.data.LongBlock;
+import org.elasticsearch.compute.operator.DriverContext;
+import org.elasticsearch.core.Releasable;
+import org.elasticsearch.core.Releasables;
+
+/**
+ * A rate grouping aggregation definition for int.
+ * This class is generated. Edit `X-RateAggregator.java.st` instead.
+ */
+@GroupingAggregator(
+    includeTimestamps = true,
+    value = {
+        @IntermediateState(name = "timestamp", type = "LONG_BLOCK"),
+        @IntermediateState(name = "value", type = "INT_BLOCK"),
+        @IntermediateState(name = "compensation", type = "DOUBLE") }
```

Comment:
Do we need a `compensation` for `int` and `long` rates? It looks a little like a Kahan sum we shouldn't need that for `int` and `long`.

I was going to ask if we need a `rate` flavor for `int` and `long` at all but I think it's right and proper to have one, at least for `long`. Lots of stuff is going to read it's values as `long`. And if we can avoid having compensation, all the better.

Reply:
> Do we need a compensation for int and long rates?

I might be missing something here. But the `compensation` is the sum of all reset values within this group's range. Perhaps, the term `compensation` is confusing? Should we consider renaming it to `totalReset`?

Reply:
Yeah! Let's rename it so I don't think it's kahan.

Reply:
I've renamed `compensation` to `reset` in https://github.com/elastic/elasticsearch/pull/106703/commits/89880d4a32dc2e7577b57d01cf7175f6c1b2f927.